app/modules/media/media_service.py

```
@@ -0,0 +1,624 @@
+
+    async def _upload_to_gcs(
+        self, storage_path: str, file_data: bytes, mime_type: str
+    ) -> None:
+        """Upload file to Google Cloud Storage"""
+        try:
+            blob = self.bucket.blob(storage_path)
+            blob.upload_from_string(file_data, content_type=mime_type)
+            logger.info(f"Successfully uploaded to GCS: {storage_path}")
+        except Exception as e:
+            logger.error(f"Failed to upload to GCS: {str(e)}")
+            raise MediaServiceError(f"Failed to upload to cloud storage: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)